Task.java

```diff
@@ -0,0 +1,69 @@
+public abstract class Task {
+    private String name;
+    private TaskStatus status;
+    private long schedule;
+    private Set<Task> dependsOn;
+
+    public abstract void runTask(EntityManager entityManager);
+
+    protected void finish(){
+        status = TaskStatus.Finished;
+        TaskManager.getInstance().taskFinished(this);
+    }
+
+    protected void reschedule(long timestamp){
+        schedule = timestamp;
+        TaskManager.getInstance().taskFinished(this);
+        TaskManager.getInstance().skeduleTask(this);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
Null checks for object reference and class fields.

Reply:
Null is present